src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -101,26 +124,120 @@ static bool queryForward(ConnectionInfo* ciP, QueryContextRequest* qcrP, QueryCo
 
 
   //
-  // 2. Render the string of the request we want to forward
+  // 2. Prepare the request to forward
+  //    - If V1: Render the payload
+  //    - If V2: Setup the URI params
   //
-  std::string  payload;
-  TIMED_RENDER(payload = qcrP->toJsonV1());
+  std::string     payload;
+  std::string     verb;
+  std::string     resource;
+  std::string     tenant       = ciP->tenant;
+  std::string     servicePath  = (ciP->httpHeaders.servicePathReceived == true)? ciP->httpHeaders.servicePath : "";
+  std::string     mimeType;
+
+  if (qcrP->providerFormat == PfJson)
+  {
+    verb      = "POST";
+    resource  = prefix + "/queryContext";
+    mimeType  = "application/json";
+
+    LM_T(LmtForward, ("Rendering payload for the forward"));
+    TIMED_RENDER(payload = qcrP->toJsonV1());
+  }
+  else
+  {
+    //
+    // NGSIv2 forward: instead of payload, URI params are used
+    //
+    std::string  extraParams;
+
+    verb      = "GET";
+    resource  = prefix + "/entities";
+
+    if (ciP->uriParam["type"] != "")
```

Comment:
I think this strategy to set the type is not going to work in the case the type comes in the payload, i.e. client request is `POST /v2/op/query` or `POST /v1/queryContext`. All data should come from `qcrP` which supposelly has been "translated" from the original request and have all information.

However, better to include in a next PR, after having some .test confirming and raising the problem. In this PR just marking this with a FIXME mark would be enough.

Reply:
I think the solution could be something similar to what is done for attrs a bit bellow:

```
    if (ciP->uriParam["attrs"] != "")
    {
    ...
    }
    else if (qcrP->attributeList.size() != 0)
```
I mean, first trying to get from URI params. If fails, try with `qcrP` info.

Reply:
Makes sense what you say.
This PR was about `GET /v2/entities` where the type will only come as uri param.
Adding FIXME for the future

Reply:
Fixed by 3c29a94